### DISCUSSION TITLE
How override route src with real ip

### DISCUSSION THREAD

**Discussion by bisontim** [acassen/keepalived#2386] [Q&A]
Hi,
I set keepalived with two nodes.
NodeA: 192.168.2.33
NodeB: 192.168.2.34

Virtual IP: 198.168.2.35

When I reach o different subnet, with the directive: 
virtual_routes {
        default via 192.168.2.1 src 192.168.2.35
    }

it works for different subnet/network, but, if I need to reach the same subnet, for example: 192.168.2.40
I obtain as source ip, the real ip 192.168.2.33 instead of 192.168.2.35

This is because:
192.168.2.0/24 dev ens192 proto kernel scope link src 192.168.2.33
How to override src ip  when I need to reach the same subnet?

Thanks a lot

Mario

**Comment by pqarmitage** [comment]
You need to add another route with a lower metric, e.g. ` ip route add 192.168.2.0/24 dev ens122 src 192.168.2.40 metric 10`

You should be able to get keepalived to add it by using a `virtual_route` so the route is only added when the VRRP instance is in maser state.
